### ISSUE TITLE
server side merge: topojson -> topojson

### ISSUE THREAD

**Issue by 015988** [topojson/topojson#211]
Hi all

-- Context:
I'm currently working on a visualization project involving a reasonable amount of georeferenced data, hosted on a postgis server. My first attempt (I'm not well versed yet in all things geo*) involved geoserver to provide wms layers that I put over a gmaps basemap (My boss is very fond of the google interface, so that needs to be on the final result). However, either from my configuration ignorance or due to the sheer amount of data, the geoserver process couldn't really handle the rendering in a reasonable time. The exact word my boss used was 'unacceptable'. :(

I'm currently trying to move the rendering to the client side, reducing the load on the server. So I have a sql query that spits a geojson that I load into the gmaps using data.addGeoJson. It works faily well, except for 2 big issues: 
- Too much data to download (my testing subset involves a very small subregion of the data and it's 25mb worth of json) 
- Each geometry is subdivided into several smaller ones, either from historical reasons or to speed up the queries. If I use ST_dump/ST_union, it solves that issue, but it takes _forever_.

AFAIK, topojson would be the solution to my problem, the same data, after simplification/merge (postgis merge) is currently amounting to around 500kb and the test I did with merge were lightning fast compared to postgis' ST_Union.

-- Request:
What I really want is to be able to run this code: 

using nodejs, on the server:

``` js
 var topology = topojson.topology(result.rows[0]); // I use more options to preserve props/ids
 topojson.merge(topology, topology.objects.XXX.geometries);
 topojson.simplify(topology,{"minimum-area": 0.00001, "coordinate-system": "cartesian"});
response.end(JSON.stringify(topology));
```

Then, on the client side:

``` js
 $.getJSON( "(nodejs call)", function( topology ) {
              map.data.addGeoJson(topojson.feature(topology, topology.objects.tcregions));}
```

-- TLDR;
merge, server side, taking a topology as argument, altering it in place (or returning it).

Full disclosure disclaimer: This might be absurd/done already, etc. I'm a noob :)

Thanks for everything (topojson is awesome. So is d3, even if i'm not using it on this bit, I'm using a whole bucketload of it on another part of the project) and thanks in advance for this request.

Fabio
(twitter: @diasf)
